numpy/distutils/system_info.py

```diff
@@ -1540,140 +1540,212 @@ def get_atlas_version(**config):
-        if not atlas_info:
-            atlas_info = get_info('atlas_blas')
-
-        accelerate_info = get_info('accelerate')
-        if accelerate_info and not atlas_info:
-            self.set_info(**accelerate_info)
-            return
+    def _calc_info_accelerate(self):
+        info = get_info('accelerate')
+        if info:
+            self.set_info(**info)
+            return True
+        return False
 
-        need_blas = 0
+    def _calc_info_blas(self):
+        # When one is finally asking for BLAS we need to report
+        # of insufficient usage of optimized BLAS
+        warnings.warn(AtlasNotFoundError.__doc__, stacklevel=2)
         info = {}
-        if atlas_info:
-            info = atlas_info
+        dict_append(info, define_macros=[('NO_ATLAS_INFO', 1)])
+
+        blas = get_info('blas')
+        if blas:
+            dict_append(info, **blas)
         else:
-            warnings.warn(AtlasNotFoundError.__doc__, stacklevel=2)
-            need_blas = 1
-            dict_append(info, define_macros=[('NO_ATLAS_INFO', 1)])
-
-        if need_blas:
-            blas_info = get_info('blas')
-            if blas_info:
-                dict_append(info, **blas_info)
-            else:
-                warnings.warn(BlasNotFoundError.__doc__, stacklevel=2)
-                blas_src_info = get_info('blas_src')
-                if not blas_src_info:
-                    warnings.warn(BlasSrcNotFoundError.__doc__, stacklevel=2)
-                    return
-                dict_append(info, libraries=[('fblas_src', blas_src_info)])
+            # Not even BLAS was found!
+            warnings.warn(BlasNotFoundError.__doc__, stacklevel=2)
+
+            blas_src = get_info('blas_src')
+            if not blas_src:
+                warnings.warn(BlasSrcNotFoundError.__doc__, stacklevel=2)
+                return False
+            dict_append(info, libraries=[('fblas_src', blas_src)])
 
         self.set_info(**info)
-        return
+        return True
+
+    def calc_info(self):
+        # Copy list so we can overwrite
+        blas_order = self.blas_order[:]
+
+        user_order = os.environ.get('NUMPY_BLAS_ORDER', None)
+        if not user_order is None:
+            # the user has requested the order of the
+            # check they are all in the available list
+            user_order = user_order.lower().split(',')
+            non_existing = []
+            for order in user_order:
+                if order not in blas_order:
+                    non_existing.append(order)
+            if len(non_existing) > 0:
+                raise ValueError("blas_opt_info user defined BLAS order has unacceptable values: {}".format(non_existing))
+            blas_order = user_order
+
+        for blas in blas_order:
+            if getattr(self, '_calc_info_{}'.format(blas))():
+                return
 
+        # possible fail-handler? Raise something?
+        # reaching this point means that nothing has been found
```

Comment:
This should be handled already by the `BlasNotFoundError` and `BlasSrcNotFoundError` usages further up, right? Admittedly those are quite weird. But can't raise anything here, or it will disable `lapack_lite` completely (which may not be a bad thing at some point, but here isn't the place to discuss that).

Reply:
Yeah.